test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/RequestSizeLimitResourceFilterTest.cs

```diff
@@ -0,0 +1,162 @@
+
+            var requestSizeLimitResoureFilter = new RequestSizeLimitResourceFilter(loggerFactory);
+            requestSizeLimitResoureFilter.Bytes = 12345;
+            var resourceExecutingContext = CreateResourceExecutingContext(new IFilterMetadata[] { requestSizeLimitResoureFilter });
+
+            // Act
+            requestSizeLimitResoureFilter.OnResourceExecuting(resourceExecutingContext);
+
+            // Assert
+            Assert.Single(sink.Writes);
+            Assert.Equal($"The filter {nameof(RequestSizeLimitResourceFilter)} could not find the feature {nameof(IHttpMaxRequestBodySizeFeature)}.", 
+                sink.Writes[0].State.ToString());
+        }
+
+        [Fact]
+        public void LogsFeatureIsReadOnly()
+        {
+            // Arrange
+            var sink = new TestSink();
+            var loggerFactory = new TestLoggerFactory(sink, enabled: true);
+
+            var requestSizeLimitResoureFilter = new RequestSizeLimitResourceFilter(loggerFactory);
+            requestSizeLimitResoureFilter.Bytes = 12345;
+            var resourceExecutingContext = CreateResourceExecutingContext(new IFilterMetadata[] { requestSizeLimitResoureFilter});
+
+            var httpMaxRequestBodySize = new TestHttpMaxRequestBodySizeFeature();
+            httpMaxRequestBodySize.IsReadOnly = true;
+            resourceExecutingContext.HttpContext.Features.Set<IHttpMaxRequestBodySizeFeature>(httpMaxRequestBodySize);
+
+            // Act
+            requestSizeLimitResoureFilter.OnResourceExecuting(resourceExecutingContext);
+
+            // Assert
+            Assert.Single(sink.Writes);
+            Assert.Equal($"The implementation of {nameof(IHttpMaxRequestBodySizeFeature)} is read-only. The attribute cannot be applied.", sink.Writes[0].State.ToString());
+        }
+
+        [Fact]
+        public void RequestSizeLimitAttribute_LogsMaxRequestBodySizeSet()
+        {
+            // Arrange
+            var sink = new TestSink();
+            var loggerFactory = new TestLoggerFactory(sink, enabled: true);
+
+            var requestSizeLimitResoureFilter = new RequestSizeLimitResourceFilter(loggerFactory);
+            requestSizeLimitResoureFilter.Bytes = 12345;
+            var resourceExecutingContext = CreateResourceExecutingContext(new IFilterMetadata[] { requestSizeLimitResoureFilter });
+
+            var httpMaxRequestBodySize = new TestHttpMaxRequestBodySizeFeature();
+            resourceExecutingContext.HttpContext.Features.Set<IHttpMaxRequestBodySizeFeature>(httpMaxRequestBodySize);
+
+            // Act
+            requestSizeLimitResoureFilter.OnResourceExecuting(resourceExecutingContext);
+
+            // Assert
+            Assert.Single(sink.Writes);
+            Assert.Equal($"The maximum request body size has been set to 12345.", sink.Writes[0].State.ToString());
+        }
+
+        [Fact]
+        public void DisableRequestSizeLimitAttribute_LogsMaxRequestBodySizeSetToNull()
```

Comment:
This is part of a completely separate test class. That `DisableRequestSizeLimitResourceFilterTest` class should be almost identical to `RequestSizeLimitResourceFilterTest` because the two filters share nothing.

Put another way, you're testing `RequestSizeLimitResourceFilter` every which way but `DisableRequestSizeLimitResourceFilter` hardly at all.